handlers/inbox-find.js

```diff
@@ -45,94 +45,266 @@ module.exports = {
-                                 options: l.options,
-                              });
-                           }
-                        }
+            const invalidInputError = new Error(
+               "Invalid Inputs. Provide either a user or a role."
+            );
+
+            cb(invalidInputError);
+
+            return;
+         }
+
+         // build our condition
+         const cond = { status: "pending" };
+
+         if (users.length) cond.or = [{ users }];
+         if (roles.length) {
+            cond.or = cond.or || [];
+            cond.or.push({ roles });
+         }
+
+         try {
+            const list = await req.retry(() =>
+               AB.objectProcessForm().model().find(cond, req)
+            );
+
+            if (list) {
+               const parsedList = [];
+
+               // make sure the .ui field is sent back as an object:
+               for (const l of list) {
+                  if (l.ui && typeof l.ui === "string") {
+                     try {
+                        l.ui = JSON.parse(l.ui);
+                     } catch (e) {
+                        let msg = `Error: UserForm[${l.uuid}].ui is not valid JSON `;
+                        req.log(msg);
+                        req.log(e);
+                        req.notify.builder(e, {
+                           context: msg,
+                           uuid: l.uuid,
+                           ui: l.ui,
+                        });
+                     }
+                  }
+
+                  if (l.options && typeof l.options === "string") {
+                     try {
+                        l.data = JSON.parse(l.options);
+                     } catch (e) {
+                        let msg = `Error: UserForm[${l.uuid}].options is not valid JSON `;
+                        req.log(msg);
+                        req.log(e);
+                        req.notify.builder(e, {
+                           context: msg,
+                           uuid: l.uuid,
+                           options: l.options,
+                        });
+                     }
+                  }
+
+                  if (!l.scopeQuery) {
+                     parsedList.push(l);
+
+                     continue;
+                  }
+
+                  try {
+                     if (typeof l.scopeQuery === "string")
+                        l.scopeQuery = JSON.parse(l.scopeQuery);
+                  } catch (e) {
+                     let msg = `Error: UserForm[${l.uuid}].scopeQuery is not valid JSON `;
+                     req.log(msg);
+                     req.log(e);
+                     req.notify.builder(e, {
+                        context: msg,
+                        uuid: l.uuid,
+                        scopeQuery: l.scopeQuery,
                      });
+                  }
+
+                  try {
+                     const dataProcessInstance = (
+                        await AB.objectProcessInstance()
```

Comment:
Any of these calls too `.model().find()`  should be wrapped in a `req.retry()` to make sure we can handle common DB connection problems.

So, for example:
```
await req.retry(() => AB.objectProcessInstance().model().find({ uuid: l.process }, req))[0]
```